lib/Transforms/IndexSwitchToIf.cpp

```diff
@@ -0,0 +1,117 @@
+//===- IndexSwitchToIf.cpp - Index switch to if-else pass ---*-C++-*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Contains the definitions of the SCF IndexSwitch to If-Else pass.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Transforms/Passes.h"
+#include "mlir/Conversion/LLVMCommon/ConversionTarget.h"
+#include "mlir/Conversion/LLVMCommon/Pattern.h"
+#include "mlir/Dialect/Func/IR/FuncOps.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/IR/OperationSupport.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+
+namespace circt {
+#define GEN_PASS_DEF_INDEXSWITCHTOIF
+#include "circt/Transforms/Passes.h.inc"
+} // namespace circt
+
+using namespace mlir;
+using namespace circt;
+
+struct SwitchToIfConversion : public OpConversionPattern<scf::IndexSwitchOp> {
+  using OpConversionPattern::OpConversionPattern;
+
+  LogicalResult
+  matchAndRewrite(scf::IndexSwitchOp switchOp, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    auto loc = switchOp.getLoc();
+
+    Region &defaultRegion = switchOp.getDefaultRegion();
+
+    Value finalResult;
+    scf::IfOp prevIfOp = nullptr;
+
+    rewriter.setInsertionPointAfter(switchOp);
+    for (size_t i = 0; i < switchOp.getCases().size(); i++) {
+      auto caseValueInt = switchOp.getCases()[i];
```

Comment:
I'd suggest initialization a variable rather than calling getCases many times.
```suggestion
    auto switchCases = switchOp.getCases();
    for (size_t i = 0; i < switchCases.size(); i++) {
      auto caseValueInt = switchCases[i];
```

Then use switchCases elsewhere as well.